core/execution/common/liquidity_provision.go

```diff
@@ -512,22 +513,28 @@ func (m *MarketLiquidity) AmendLiquidityProvision(
 	}
 
 	// if pending commitment is being decreased then release the bond collateral
-	existingAmendment := m.liquidityEngine.PendingProvisionByPartyID(party)
+
+	var proposedCommitmentVariation num.Decimal
+
 	if existingAmendment != nil && !lpa.CommitmentAmount.IsZero() && lpa.CommitmentAmount.LT(existingAmendment.CommitmentAmount) {
 		amountToRelease := num.UintZero().Sub(existingAmendment.CommitmentAmount, lpa.CommitmentAmount)
-		_, err := m.releasePendingBondCollateral(ctx, amountToRelease, party)
-		if err != nil {
+		if err := m.releasePendingBondCollateral(ctx, amountToRelease, party); err != nil {
 			m.log.Debug("could not submit update bond for lp amendment",
 				logging.PartyID(party),
 				logging.MarketID(m.marketID),
 				logging.Error(err))
 
 			return err
 		}
+		proposedCommitmentVariation = existingAmendment.CommitmentAmount.ToDecimal().Sub(lpa.CommitmentAmount.ToDecimal())
+	} else {
+		if lp != nil {
+			proposedCommitmentVariation = lp.CommitmentAmount.ToDecimal().Sub(lpa.CommitmentAmount.ToDecimal())
+		} else {
+			proposedCommitmentVariation = existingAmendment.CommitmentAmount.ToDecimal().Sub(lpa.CommitmentAmount.ToDecimal())
+		}
```

Comment:
why do we need the ToDecimal conversions her?

Reply:
I guess because it could be negative